packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksToolbar.tsx

```diff
@@ -187,7 +226,18 @@ const BlocksDropdown = () => {
       Transforms.select(editor, Editor.start(editor, [0, 0]));
     }
 
-    // Let the block handle the Slate conversion logic
+    // If selection is a list-item, toggle format
+    const currentListEntry = Editor.above(editor, {
+      match: (node) => !Editor.isEditor(node) && node.type === 'list',
+    });
+
+    if (currentListEntry && ['list-ordered', 'list-unordered'].includes(optionKey)) {
+      const format = optionKey.split('-')[1] as Block<'list'>['format'];
+      handleToggleList(editor, currentListEntry, format);
+      return;
+    }
+
+    // Let the block handle the Slate conversion logic except list
```

Comment:
why do we need this exception? why not have the converting logic in handleConvert like all other blocks?

I think it's important to centralize the converting logic as the conversions can happen from different places

Reply:
My bad, actually `handleConvert` do handle **any block to list** conversion, condition I added above is to check if its already a list block then just toggle the format. I updated the comment.